agenta-web/src/components/Scripts/GlobalScripts.tsx

```diff
@@ -1,24 +1,24 @@
-import {useState, useEffect, useCallback, ComponentType, useRef} from "react"
+import {useState, useEffect, useCallback, ComponentType} from "react"
 import Head from "next/head"
 import {isDemo} from "@/lib/helpers/utils"
 import {dynamicComponent} from "@/lib/helpers/dynamic"
 
 const GlobalScripts = () => {
     const [CloudScripts, setCloudScripts] = useState<ComponentType | null>(null)
-    const isLoading = useRef(false)
+    const [isLoading, setIsLoading] = useState(false)
 
     const initializeScripts = useCallback(() => {
         const Scripts = dynamicComponent("Scripts/assets/CloudScripts")
-        setCloudScripts(() => Scripts)
+        setCloudScripts((prev: any) => prev || Scripts)
```

Comment:
```suggestion
        setCloudScripts((prev: any) => prev ?? Scripts)
```

didn't pay attention to initial value